web/src/compose_validate.ts

```diff
@@ -837,40 +875,60 @@ export function check_overflow_text($container: JQuery): number {
     return text.length;
 }
 
-export function validate_message_length($container: JQuery): boolean {
+export function validate_message_length($container: JQuery, show_banner = true): boolean {
     const $textarea = $container.find<HTMLTextAreaElement>(".message-textarea");
     // Match the behavior of compose_state.message_content of trimming trailing whitespace
     const text = $textarea.val()!.trimEnd();
     const message_too_long_for_compose = text.length > realm.max_message_length;
     set_message_too_long_for_compose(message_too_long_for_compose);
     if (message_too_long_for_compose) {
-        $textarea.addClass("flash");
-        setTimeout(() => $textarea.removeClass("flash"), 1500);
+        if (show_banner) {
+            $textarea.addClass("flash");
+            setTimeout(() => $textarea.removeClass("flash"), 1500);
+        }
         return false;
     }
     return true;
 }
 
-export function validate(scheduling_message: boolean): boolean {
+function report_validation_error(
+    message: string,
+    classname: string,
+    $container: JQuery,
+    $bad_input: JQuery,
+    show_banner: boolean,
+    precursor?: () => void,
+): void {
+    if (show_banner) {
+        if (precursor) {
+            precursor();
+        }
+        compose_banner.show_error_message(message, classname, $container, $bad_input);
+    }
+}
+export function validate(scheduling_message: boolean, show_banner = true): boolean {
```

Comment:
Is `show_banner` ever anything but `true`? I wonder if this is a stale parameter?